kmod/src/kernelcompat.c

```diff
@@ -21,3 +21,33 @@ int kc_shrink_wrapper(struct shrinker *shrink, struct shrink_control *sc)
 	return min_t(unsigned long, nr, INT_MAX);
 }
 #endif
+
+#ifndef KC_CURRENT_TIME_INODE
+struct timespec64 kc_current_time(struct inode *inode)
+{
+	struct timespec64 now;
+	unsigned gran;
+
+	getnstimeofday64(&now);
+
+	if (unlikely(!inode->i_sb)) {
```

Comment:
Why does this have a copy of timespec_truncate?  can't we just call it?  It looks like it's consistently exported.

Reply:
This is a port of timespec64_trunc, which truncates a timespec64. The available timespec_trunc() function truncates a struct timespec. Hence the "copy" - the types differ.